unity/Assets/Tests/EditModeTests/ContentValidation/PcSceneValidationTests.cs

```diff
@@ -371,5 +375,76 @@ public void SceneHasGlobalEntities()
             var gameObject = FindObjectByName(objectNameUnderTest);
             AssertGameObjectIsActive(gameObject);
         }
+
+        [Test, Description("MonoBehaviours must not have any missing target objects or methods for UnityEvents.")]
+        public void SceneHasNoMissingUnityEventMethods()
+        {
+            // Based on https://gist.github.com/AaronV/3fd7cc22039cf34f536ab98db47d044a
+            // and https://stackoverflow.com/questions/42784338/unity-missing-warning-when-button-has-missing-onclick/42788400
+
+            List<string> errors = new List<string>();
+
+            // Iterate over all MonoBehaviours
+            MonoBehaviour[] monoBehavioursInScene = Resources.FindObjectsOfTypeAll<MonoBehaviour>();
+            foreach(MonoBehaviour monoBehaviour in monoBehavioursInScene)
+            {
+                // Check all fields whether they are a UnityEvent
+                Type monoBehaviourType = monoBehaviour.GetType();
+                FieldInfo[] fields = monoBehaviourType.GetFields(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+                foreach (FieldInfo field in fields)
+                {
+                    if (typeof(UnityEvent).IsAssignableFrom(field.FieldType))
+                    {
+                        UnityEvent unityEvent = field.GetValue(monoBehaviour) as UnityEvent;
+                        // Check all persistent (assigned via the Inspector) UnityEvent listeners
+                        for (int persistentEventCountIndex = 0; persistentEventCountIndex < unityEvent.GetPersistentEventCount(); persistentEventCountIndex++)
+                        {
+                            // Assert event target object is not null
+                            UnityEngine.Object eventTargetObject = unityEvent.GetPersistentTarget(persistentEventCountIndex);
+                            if (eventTargetObject == null)
+                            {
+                                errors.Add($"The UnityEvent of {monoBehaviourType.Name} \"{monoBehaviour.name}\" called \"{field.Name}\" " +
+                                    $"has an event target object that is null (index {persistentEventCountIndex}).");
+                                continue;
+                            }
+
+                            // Assert event target object Type is not null
+                            string eventTargetObjectFullName = eventTargetObject.GetType().AssemblyQualifiedName;
```

Comment:
AssemblyQualifiedName can be null if the current instance represents a generic type parameter.

Reply:
Not sure I understand this fully;
According to the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname?view=net-8.0) the AssemblyQualifiedName can return null if the type represents a generic type parameter. (like you said)
However, the eventTargetObject should never be a generic Type T? I.e. if I understand correctly, if you have e.g. MyGenericClass<T> myObject and then do typeof(T).AssemblyQualifiedName that might be null? but myObject.GetType().AssemblyQualifiedName should not be null?
But I am not 100% sure I understand that fully, so I'll add a null check, hope that's ok